### PR TITLE
clean front

### PR DESCRIPTION
**Pull request by AugustinMauroy**
## Description

Update the front-end code.

- add type def for node (used on vite config)
- strict separation between function and type def for import
- remove `import * as React form 'react'` because it's wrap by vite

### CODE REVIEW

**Comment by Fredkiss3** [comment]
`import * as React form 'react'` is a choice to distinguish between React hooks and custom hooks, also We don't use `FC` instead we prefer regular functions.

File: frontend/src/components/helper/auth-redirect.tsx [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+export function withAuthRedirect(WrappedComponent: ComponentType) {
   return function AuthRedirectWrapper(props: any) {
     const navigate = useNavigate();
 
     const query = useAuthUser();
 
-    if (query.isLoading) {
-      return <Loader />;
-    }
+    if (query.isLoading) return <Loader />;
```
Comment (Fredkiss3) [inline frontend/src/components/helper/auth-redirect.tsx:13]: This is deliberate choice, inlined conditions are reserved for the JSX, we prefer to put conditions in braces if possible.

File: frontend/src/components/nav-link.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
-import { Link, type LinkProps } from "@tanstack/react-router";
-import React from "react";
+import { Link } from "@tanstack/react-router";
+import { forwardRef } from "react";
+
+import type { LinkProps } from "@tanstack/react-router";
+import type { ElementRef } from "react";
```
Comment (Fredkiss3) [inline frontend/src/components/nav-link.tsx:5]:
```suggestion
import * as React from "react";
```

`import * as React form 'react'` is a choice to distinguish between React hooks and custom hooks

File: frontend/src/components/nav-link.tsx [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 export type NavLinkProps = Omit<LinkProps, "to" | "ref" | "activeProps"> & {
   href: string;
 };
 
-export const NavLink = React.forwardRef<
-  React.ElementRef<typeof Link>,
-  NavLinkProps
->(function NavLink({ href, ...props }, ref) {
-  return (
+export const NavLink = forwardRef<ElementRef<typeof Link>, NavLinkProps>(
```
Comment (Fredkiss3) [inline frontend/src/components/nav-link.tsx:11]:
```suggestion
export const NavLink = React.forwardRef<React.ElementRef<typeof Link>, NavLinkProps>(
```

Idem, this is a choice

File: frontend/src/components/meta-title.tsx [hunk 4]
```diff
@@ -1,3 +1,9 @@
-export function MetaTitle({ title }: { title: string }) {
-  return <title>{`${title} | ZaneOps`}</title>;
-}
+import type { FC } from "react";
+
+type MetaTitleProps = {
+  title: string;
+};
+
+export const MetaTitle: FC<MetaTitleProps> = ({ title }) => (
```
Comment (Fredkiss3) [inline frontend/src/components/meta-title.tsx:7]: We don't use `FC` we prefer regular functions instead, this is a deliberate choice.

**Review by Fredkiss3 — CHANGES_REQUESTED** [review]
Some changes : 
- `import * as React from 'react'` is a deliberate choice to distinguish between react hooks and custom hooks and since React is always bundled fully, it doesn't have any effect on bundling
- we don't use `FC`, this is deliberate choice, and we prefer regular functions with explicit arguments.

File: frontend/src/components/meta-title.tsx (hunk 4 above)
Comment (AugustinMauroy) [reply frontend/src/components/meta-title.tsx:7]: `FC` mean Functional component it's cleaner that simple function and type check can "test" it

Comment (Fredkiss3) [reply frontend/src/components/meta-title.tsx:7]: This is subjective, but i find writing functions cleaner, we don't really to test them with the type check, if it is used as a React node, TypeScript will check it. Please not also that React jsx nodes are very permissive, so in practice this doesn't add very big improvements.

Comment (Fredkiss3) [reply frontend/src/components/meta-title.tsx:7]: I already know what `FC` is and have already used it in previous projects, but functions are cleaner (for me).

**Comment by Fredkiss3** [comment]
Also @AugustinMauroy  can you keep the default PR template, it's important thank you 🙏🏾

**Comment by AugustinMauroy** [comment]
Should I close this issue? 'cause basically what I propose it's contradictory to the project direction.

**Comment by Fredkiss3** [comment]
hm... yeah sorry 🙏🏾

**Pull request closed without merging**